app/logic/Auth/OAuth2.ts

```diff
@@ -222,25 +235,42 @@ export class OAuth2 extends Observable {
     return this.accessToken;
   }
 
-  getAuthURL(doneURL?: URLString): URLString {
-    this.verificationToken = Math.random().toString().slice(2);
+  async getAuthURL(doneURL?: URLString): Promise<URLString> {
     this.authDoneURL = doneURL ?? this.authDoneURL; // needed for getAccessTokenFromAuthCode()
-    return this.authURL + "?" + new URLSearchParams({
+    let params = new URLSearchParams({
       client_id: this.clientID,
       response_type: "code",
       redirect_uri: doneURL ?? this.authDoneURL,
-      response_mode: "query",
       scope: this.scope,
-      state: this.verificationToken,
-      login_hint: this.account.username,
```

Comment:
Why do you remove those?
* state is required
* login_hint is required for Microsoft
* response_mode is needed as well, I would presume

Did you test this with Microsoft accounts, that you don't regress anything? I think these changes would break Microsoft login (EWS with Office365 and OAuth2)